pkg/cmd/join/cmd.go

```diff
@@ -55,5 +55,9 @@ func NewCmd(clusteradmFlags *genericclioptionsclusteradm.ClusteradmFlags, stream
 		"If true, the installed klusterlet agent will be starting the cluster registration process by "+
 			"looking for the internal endpoint from the public cluster-info in the hub cluster instead of from --hub-apiserver.")
 	cmd.Flags().BoolVar(&o.wait, "wait", false, "If true, running the cluster registration in foreground.")
+	cmd.Flags().StringVarP(&o.mode, "mode", "m", "default", "mode to deploy klusterlet, can be default or hosted")
+	cmd.Flags().StringVar(&o.managedKubeconfigFile, "managed-cluster-kubeconfig", "", "To specify the directory to external managed cluster kubeconfig in hosted mode")
+	cmd.Flags().StringVar(&o.klusterletName, "klusterlet-name", "klusterlet", "To specify klusterlet name(the klusterlet will deploy in the namespace with same name of klusterlet name) in hosted mode")
```

Comment:
maybe we can just use cluster name as default value for klusterletname and klusterletnamespace. @zhujian7 would you like to take a look

Reply:
if we use the cluster name and the hosting cluster is the hub cluster, not sure if there will be any issue when detaching the cluster.